src/pages/VaccinePage.js

```diff
@@ -1,125 +1,196 @@
+              size="ms"
+              containerStyle={localStyles.fridgeDetail}
+              Icon={<BatteryIcon color={DARKER_GREY} />}
+            >
+              {`${batteryLevel}%`}
+            </TextWithIcon>
+            <IconButton
+              Icon={<ChevronRightIcon color={BLACK} />}
+              onPress={() => toFridgeDetail(fridge)}
+              containerStyle={localStyles.iconButton}
+            />
+          </FlexRow>
+        );
+      })}
+    </Paper>
   );
 };
-
-const mapStateToProps = state => {
-  const { fridge } = state;
-  const { fridges, selectedFridge } = fridge;
-  const breaches = selectBreaches(state);
-  const isAdmin = selectCurrentUserIsAdmin(state);
-  const { minLine, maxLine } = selectMinAndMaxLogs(state);
-  const { minDomain, maxDomain } = selectMinAndMaxDomains(state);
-  return {
-    isAdmin,
-    minLine,
-    maxLine,
-    minDomain,
-    maxDomain,
-    fridges,
-    selectedFridge,
-    breaches,
-  };
+FridgeDisplay.defaultProps = {
+  fridge: null,
+};
+FridgeDisplay.propTypes = {
+  fridge: PropTypes.shape({
+    id: PropTypes.string,
+    description: PropTypes.string,
+    sensors: PropTypes.object,
+  }),
+  blinkSensor: PropTypes.func.isRequired,
+  toFridgeDetail: PropTypes.func.isRequired,
+  toEditSensorPage: PropTypes.func.isRequired,
 };
 
-const mapDispatchToProps = dispatch => ({
-  onViewAdminPage: () => dispatch(gotoVaccineAdminPage()),
-  onSelectFridge: fridge => dispatch(FridgeActions.select(fridge)),
-  onOpenBreachModal: breachId => {
-    dispatch(BreachActions.viewFridgeBreach(breachId));
-  },
-});
-
+export const VaccinePageComponent = ({ fridges, toNewSensorPage, ...dispatchers }) => (
+  <DataTablePageView style={{ paddingHorizontal: 20, paddingVertical: 30 }}>
+    <FlexRow style={{ justifyContent: 'flex-end', marginBottom: 10 }}>
+      <PageButton text={buttonStrings.add_sensor} onPress={toNewSensorPage} />
+    </FlexRow>
+    <AfterInteractions placeholder={null}>
+      <Animatable.View animation="fadeIn" duration={500} useNativeDriver>
+        <FlatList
+          renderItem={({ item }) => <FridgeDisplay fridge={item} {...dispatchers} />}
+          data={fridges}
+        />
+      </Animatable.View>
+    </AfterInteractions>
+  </DataTablePageView>
+);
 VaccinePageComponent.defaultProps = {
-  breaches: [],
-  selectedFridge: null,
+  fridges: null,
 };
-
 VaccinePageComponent.propTypes = {
-  selectedFridge: PropTypes.object,
-  minLine: PropTypes.array.isRequired,
-  maxLine: PropTypes.array.isRequired,
-  breaches: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
-  maxDomain: PropTypes.number.isRequired,
-  minDomain: PropTypes.number.isRequired,
-  fridges: PropTypes.oneOfType([PropTypes.array, PropTypes.object]).isRequired,
-  onSelectFridge: PropTypes.func.isRequired,
-  onOpenBreachModal: PropTypes.func.isRequired,
-  onViewAdminPage: PropTypes.func.isRequired,
-  isAdmin: PropTypes.bool.isRequired,
+  fridges: PropTypes.object,
```

Comment:
Hm when the page is first loaded for me (before setting up anything), I'm getting a complaint fridges is being supplied as an array to the VaccinePage. I think `FridgeReducer` has an array default for the initial state that should maybe be converted to an object? (`locationsWithASensor`)

Reply:
Or maybe it is supposed to allow arrays. The name fridges does suggest an array but now I think I confused myself.

Reply:
It is being passed a realm collection which has the shape:

```
{ "0": someFridge, "1": someOtherFridge } 
```

which acts as an array but is technically an object!

Reply:
Right, thanks that makes a lot more sense hah. Must just be the initialState combined with my setup causing the warning then!

Reply:
d18f503